corehq/apps/ota/utils.py

```diff
@@ -131,16 +137,29 @@ def get_restore_user(domain, couch_user, as_user):
         a restore user for. If None will get restore of the couch_user.
     :returns: An instance of OTARestoreUser
     """
+    restore_user = None
     if couch_user.is_commcare_user():
         restore_user = couch_user.to_ota_restore_user()
     elif (couch_user.is_web_user() and as_user is not None):
-        username = as_user.split('@')[0]
-        domain = as_user.split('@')[1]
-        if username != couch_user.raw_username and domain == domain:
-            commcare_user = CommCareUser.get_by_username('{}.commcarehq.org'.format(as_user))
-            if not commcare_user:
+        if '@' in as_user:
+            _, user_domain = as_user.split('@')
+        else:
+            user_domain = None
+        # Likely a mobile worker because username contains domain
+        if domain == user_domain:
+            user = CommCareUser.get_by_username('{}.commcarehq.org'.format(as_user))
+            if not user:
                 return None
-            restore_user = commcare_user.to_ota_restore_user()
+            restore_user = user.to_ota_restore_user()
+        # Likely a web user
+        else:
+            user = WebUser.get_by_username(as_user)
+            if not user:
+                return None
+            elif domain not in user.domains:
```

Comment:
what's the use case for one web user restoring as another user? it seems like maybe it's be clearer to just allow web users to restore as themselves (assuming that's what's happening)